File: Combination_Samples/MVC_and_WebForms/Demo/Bridge/output/Html5App.js
Bridge.Class.define('Demo.Html5App', {
    statics: {
        config:  {
            init: function () {
                Bridge.on('DOMContentLoaded', document, this.main);
            }
        },
        main: function () {
            document.body.style.padding = "20px";
            var form = Bridge.merge(document.createElement('form'), {
                className: "col-md-6"
            } );
            var panel = Demo.Html5App.createPanel();
            form.appendChild(panel);
            document.body.appendChild(form);
            Demo.Html5App.updateButton_Click(null);
        },
        createPanel: function () {
            var panel = Bridge.merge(document.createElement('div'), {
                className: "panel panel-default"
            } );
            var header = Demo.Html5App.createPanelHeader();
            var body = Demo.Html5App.createPanelBody();
            var footer = Demo.Html5App.createPanelFooter();
            panel.appendChild(header);
            panel.appendChild(body);
            panel.appendChild(footer);
            return panel;
        },
        createPanelHeader: function () {
            var header = Bridge.merge(document.createElement('div'), {
                className: "panel-heading logo"
            } );
            var title = Bridge.merge(document.createElement('h1'), {
                innerHTML: "Bridge.NET HTML5 Demo"
            } );
            header.appendChild(title);
            return header;
        },
        createPanelBody: function () {
            var body = Bridge.merge(document.createElement('div'), {
                className: "panel-body"
            } );
            var name = Demo.Html5App.createFormField("Name", "user");
            var email = Demo.Html5App.createFormField("Email", "globe");
            var message = Demo.Html5App.createFormField("Message", "pencil");
            var dateTime = Demo.Html5App.createDateTimeField();
            body.appendChild(name);
            body.appendChild(email);
            body.appendChild(message);
            body.appendChild(dateTime[0]);
            body.appendChild(dateTime[1]);
            return body;
        },
        createPanelFooter: function () {
            var footer = Bridge.merge(document.createElement('div'), {
                className: "panel-footer text-right"
            } );
            var button = Bridge.merge(document.createElement('input'), {
                type: "submit", 
                value: "Submit", 
                className: "btn btn-success", 
                formAction: Demo.Config.SUBMIT_URL, 
                formMethod: "POST"
            } );
            footer.appendChild(button);
            return footer;
        },
        createFormField: function (name, glyph) {
            var div = Bridge.merge(document.createElement('div'), {
                className: "input-group", 
                style: {
                    marginBottom: "10px"
                }
            } );
            var span = Bridge.merge(document.createElement('span'), {
                className: "glyphicon glyphicon-" + glyph + " input-group-addon"
            } );
            var input;
            var placeholder = name + "...";
            if (name === "Message") {
                input = Bridge.merge(document.createElement('textarea'), {
                    name: name.toLowerCase(), 
                    placeholder: placeholder, 
                    required: true, 
                    className: "form-control"
                } );
            }
            else  {
                input = Bridge.merge(document.createElement('input'), {
                    type: "text", 
                    name: name.toLowerCase(), 
                    placeholder: placeholder, 
                    required: true, 
                    className: "form-control"
                } );
            }
            div.appendChild(span);
            div.appendChild(input);
            return div;
        },
        createDateTimeField: function () {
            var label = Bridge.merge(document.createElement('label'), {
                htmlFor: "dateTimeInput", 
                innerHTML: "Server Date and Time:"
            } );
            var div = Bridge.merge(document.createElement('div'), {
                className: "input-group"
            } );
            var spanPrefix = Bridge.merge(document.createElement('span'), {
                className: "input-group-addon glyphicon glyphicon-time"
            } );
            var spanSuffix = Bridge.merge(document.createElement('span'), {
                className: "input-group-btn"
            } );
            var button = Bridge.merge(document.createElement('button'), {
                type: "button", 
                className: "btn btn-primary", 
                innerHTML: "<span class=\"glyphicon glyphicon-refresh\"></span>", 
                onclick: Demo.Html5App.updateButton_Click
            } );
            var input = Bridge.merge(document.createElement('input'), {
                id: "dateTimeInput", 
                type: "text", 
                className: "form-control", 
                placeholder: "Click update...", 
                readOnly: true, 
                name: "datetime"
            } );
            spanSuffix.appendChild(button);
            div.appendChild(spanPrefix);
            div.appendChild(input);
            div.appendChild(spanSuffix);
            return [label, div];
        },
        updateButton_Click: function (e) {
            var request = new XMLHttpRequest();
            var onLoad = function () {
                var val = JSON.parse(request.responseText);
                var dateTime = Bridge.Date.parse(val);
                document.getElementById("dateTimeInput").value = Bridge.Date.format(dateTime, "yyyy-MM-dd hh:mm:ss");
            };
            request.onload = onLoad;
            request.open("GET", "/Data/GetServerTime?" + new Date().getTime(), true);
            request.send();
        }
    }
});

File: Combination_Samples/MVC_and_WebForms/Demo/Html5App.cs
using System;
using Bridge;
using Bridge.Html5;

namespace Demo
{
    [FileName("Html5App.js")]
    public class Html5App
    {
        [Ready]
        public static void Main()
        {
            Document.Body.Style.Padding = "20px";

            var form = new HTMLFormElement
            {
                ClassName = "col-md-6"
            };

            var panel = Html5App.CreatePanel();
            form.AppendChild(panel);

            Document.Body.AppendChild(form);

            UpdateButton_Click(null);
        }

        public static HTMLDivElement CreatePanel()
        {
            var panel = new HTMLDivElement
            {
                ClassName = "panel panel-default"
            };

            var header = Html5App.CreatePanelHeader();
            var body = Html5App.CreatePanelBody();
            var footer = Html5App.CreatePanelFooter();

            panel.AppendChild(header);
            panel.AppendChild(body);
            panel.AppendChild(footer);

            return panel;
        }

        public static HTMLDivElement CreatePanelHeader()
        {
            var header = new HTMLDivElement
            {
                ClassName = "panel-heading logo"
            };

            var title = new HTMLHeadingElement
            {
                InnerHTML = "Bridge.NET HTML5 Demo"
            };

            header.AppendChild(title);

            return header;
        }

        public static HTMLDivElement CreatePanelBody()
        {
            var body = new HTMLDivElement
            {
                ClassName = "panel-body"
            };

            var name = Html5App.CreateFormField("Name", "user");
            var email = Html5App.CreateFormField("Email", "globe");
            var message = Html5App.CreateFormField("Message", "pencil");
            Element[] dateTime = Html5App.CreateDateTimeField();

            body.AppendChild(name);
            body.AppendChild(email);
            body.AppendChild(message);
            body.AppendChild(dateTime[0]);
            body.AppendChild(dateTime[1]);

            return body;
        }

        public static HTMLDivElement CreatePanelFooter()
        {
            var footer = new HTMLDivElement
            {
                ClassName = "panel-footer text-right"
            };

            var button = new  HTMLInputElement
            {
                Type = InputType.Submit,
                Value = "Submit",
                ClassName = "btn btn-success",
                FormAction = Config.SUBMIT_URL,
                FormMethod = "POST"
            };

            footer.AppendChild(button);

            return footer;
        }

        public static HTMLDivElement CreateFormField(string name, string glyph)
        {
            var div = new HTMLDivElement
            {
                ClassName = "input-group",
                Style =
                {
                    MarginBottom = "10px"
                }
            };

            var span = new HTMLSpanElement
            {
                ClassName = "glyphicon glyphicon-" + glyph + " input-group-addon"
            };

            Element input;
            var placeholder = name + "...";

            if (name == "Message")
            {
                input = new HTMLTextAreaElement
                {
                    Name = name.ToLowerCase(),
                    Placeholder = placeholder,
                    Required = true,
                    ClassName = "form-control"
                };
            }
            else
            {
                input = new  HTMLInputElement
                {
                    Type = InputType.Text,
                    Name = name.ToLowerCase(),
                    Placeholder = placeholder,
                    Required = true,
                    ClassName = "form-control"
                };
            }

            div.AppendChild(span);
            div.AppendChild(input);

            return div;
        }

        public static Element[] CreateDateTimeField()
        {
            var label = new HTMLLabelElement
            {
                HtmlFor = "dateTimeInput",
                InnerHTML = "Server Date and Time:"
            };

            var div = new HTMLDivElement
            {
                ClassName = "input-group"
            };

            var spanPrefix = new HTMLSpanElement
            {
                ClassName = "input-group-addon glyphicon glyphicon-time"
            };

            var spanSuffix = new HTMLSpanElement
            {
                ClassName = "input-group-btn"
            };

            var button = new HTMLButtonElement
            {
                Type = ButtonType.Button,
                ClassName = "btn btn-primary",
                InnerHTML = "<span class=\"glyphicon glyphicon-refresh\"></span>",
                OnClick = Html5App.UpdateButton_Click
            };

            var input = new  HTMLInputElement
            {
                Id = "dateTimeInput",
                Type = InputType.Text,
                ClassName = "form-control",
                Placeholder = "Click update...",
                ReadOnly = true,
                Name = "datetime"
            };

            spanSuffix.AppendChild(button);

            div.AppendChild(spanPrefix);
            div.AppendChild(input);
            div.AppendChild(spanSuffix);

            return new Element[] { label, div };
        }

        public static void UpdateButton_Click(Event e)
        {
            var request = new XMLHttpRequest();

            Action<Event> onLoad = delegate
            {
                var val = JSON.Parse(request.ResponseText).As<string>();
                var dateTime = DateTime.Parse(val);

                Document.GetElementById< HTMLInputElement>("dateTimeInput").Value = dateTime.ToString("yyyy-MM-dd hh:mm:ss");
            };

            request.OnLoad = onLoad;
            request.Open("GET", Config.GET_SERVER_TIME_URL + "?" + DateTime.Now.ToString(), true);
            request.Send();
        }
    }
}